### ISSUE TITLE
version 3.0.0？

### ISSUE THREAD

**Issue by xiao-k233** [nothingstopsme/AX88179_178A_Linux_Driver#3]
![屏幕截图_20240310_144949](https://github.com/nothingstopsme/AX88179_178A_Linux_Driver/assets/58621432/dfef08c6-ee48-45be-a991-42c62167b718)

**Comment by frittentheke** [comment]
@xiao-k233 got a link?

**Comment by xiao-k233** [comment]
now it updated
3.1.0[link](https://www.asix.com.tw/en/product/USBEthernet/Super-Speed_USB_Ethernet/AX88179A)

**Comment by nothingstopsme** [comment]
Thank you for the update, @xiao-k233 .

It looks like the latest release, which appears to be a major revision compared with the version this project is based on,
has fixed the bugs. I could even build it, despite some build warnings, on a machine running kernel 6.5
and ran an iperf test for a few hours without any connection issues.
(Note: the test was run at usb2 speed (high speed), since the test machine I used is quite picky about usb3 cables/connectors, 
and xhci_hcd can only detect my adaptor as a usb2 device)

I'll put a note on the readme page to suggest considering the adoption of this new official version.

**Comment by gserl** [comment]
I'll leave this message here for posterity. 

TLDR; seems this newer ASIX driver series (version 3.x.x) is by far the most stable option for any USB-to-Ethernet adapter using their ax88179 or related chip. And they still can't handle it when the heat gets turned up to max!

I'm on version 3.3.0 of the official ASIX driver released several days ago. This kernel module is named ax_usb_nic. I'm running kernel 6.8.12 on debian unstable. So far the Ethernet adapter (mine's the Trendnet USB3-1Gbps version2) is uncharacteristically stable at light to moderately intense usage and browser button-slamming. Had to blacklist the older ax88179_178a module and issue an `update-initramfs -u` to prevent both modules from loading at the same time. I think you're right about the line noise issue or something, when I ran Speedtest++ from the command line, it got 1/2 way through this very intensive test (previously it never got even that far) and then hung. No receive/transmit activity, and syslog is as usual spammed with a loop of "net_ratelimit callbacks suppressed" messages. The NIC lights continue blinking (good sign). A simple unplug/replug of the adapter remedied things (also unlike with the previous driver where it was hit or miss if a replug alone would do the trick). Trying to heavily saturate the NIC still crashes it. I'm using a new Cat5e network cable (15-foot) decently insulated and running to a small unmanaged Netgear switch. Anyway, new ASIX driver is more solid than any other I've tried, but it still cannot handle full maximum saturation without choking. Didn't try torrenting but I imagine it would choke on that too on a heavy torrent, probably anytime you're getting a crazy lot of concurrent incoming and outgoing connections and its probably due to the electrical noise sensitivities of the chip (exacerabated with heat as it gets fairly hot to the touch) that was talked about here.

**Comment by gserl** [comment]
Installed newest (v 3.3.0) official ASIX driver on Debian (kernel 6.8.12). Both your out-of-tree and the official Linux kernel module were very troublesome, the Linux kernel one of course more-so. This latest ASIX driver is the most stable of any I've used, but still slow (WiFi is much faster on the laptop) and it cannot handle anything more than light web browsing, repo updates, streaming audio, and single file downloads without choking. And not all at the same time either. Forget doing any really network-intensive things like Speedtest++ runs, torrenting, or essentially anything that makes the adapter get hot (my Trendnet TU3-ETG v 2.0R gets hot easily). In these circumstances it will hang (NIC lights remain blinking, no receive/transmit capability), syslog will spam `net_ratelimit callbacks suppressed` messages. Luckily though with the new driver, a simple unplug/replug fixes it. Had to blacklist the older ax88179_178a driver and issue a `# update-initramfs -u`. I'm using a new-ish 15-ft Cat5e cable with decent (but not ultra-premium) insulation running to a small unmanaged Netgear switch. I suspect the chipset cannot handle heat and the electrical noise heat induces, I'm almost tempted to crack the plastic casing open and stick a fan on it.
